recipes/netpbm/all/conanfile.py

```diff
@@ -0,0 +1,238 @@
+        elif is_apple_os(self):
+            if self.options.shared:
+                return "dylib", "dylib"
+            else:
+                return "unixstatic", "a"
+        else:
+            if self.options.shared:
+                return "unixshared", "so"
+            else:
+                return "unixstatic", "a"
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+
+        tc = AutotoolsToolchain(self)
+        config = []
+        lib_type, suffix = self._libtype_and_suffix
+        config.append(f"NETPBMLIBTYPE={lib_type}")
+        config.append(f"NETPBMLIBSUFFIX={suffix}")
+        config.append(f"STATICLIB_TOO={'N' if self.options.shared else 'Y'}")
+        config.append(f"WANT_SSE={'Y' if self.settings.arch in ['x86', 'x86_64'] else 'N'}")
+        config.append("DEFAULT_TARGET=nonmerge")
+        config.append("LDRELOC=ld --reloc")
+
+        if is_apple_os(self):
+            config.append("LDSHLIB=-shared -dynamiclib -install_name @rpath/libnetpbm.dylib")
+
+        def _configure_dependency(name, pkg, lib):
+            cpp_info = self.dependencies[pkg].cpp_info.aggregated_components()
+            # The file does not need to exist. The build script simply strips 'lib*.a' from the paths.
+            lib_path = os.path.join(cpp_info.libdir, f"lib{lib}.a")
+            config.append(f"{name}LIB={lib_path}")
+            config.append(f"{name}HDR_DIR={cpp_info.includedir}")
+
+        def _use_pkgconfig(name):
+            config.append(f"{name}LIB=USE_PKGCONFIG.a")
+            config.append(f"{name}HDR_DIR=USE_PKGCONFIG.a")
+
+        if self.options.tools:
+            _configure_dependency("TIFF", "libtiff", "tiff")
+            _configure_dependency("PNG", "libpng", "png")
+            _configure_dependency("JPEG", str(self.options.with_libjpeg), "jpeg")
+            _configure_dependency("JASPER", "jasper", "jasper")
+            _configure_dependency("JBIG", "jbig", "jbig")
+            if self.options.get_safe("with_x11"):
+                _use_pkgconfig("X11")
+
+        cflags = list(tc.cflags)
+        if self.settings.build_type == "Debug":
+            cflags += ["-g"]
+        else:
+            # Match flags set by ./configure
+            cflags += ["-O3", "-ffast-math", "-fno-finite-math-only"]
+        if self.options.get_safe("fPIC", True):
+            cflags.append("-fPIC")
+        if is_apple_os(self):
+            # For vasprintf
+            cflags.append("-D_DARWIN_C_SOURCE")
+        config.append(f"CFLAGS={' '.join(cflags)}")
+
+        tc.make_args.extend(config)
+        tc.generate()
+
+        deps = PkgConfigDeps(self)
+        deps.generate()
+
+    def _patch_sources(self):
+        # Fix FILE not being defined in jpeglib.h
+        replace_in_file(self, os.path.join(self.source_folder, "converter", "other", "cameratopam", "camera.c"),
+                        "#include <jpeglib.h>", "#include <stdio.h>\n#include <jpeglib.h>")
```

Comment:
HAs this been reported upstream?

Reply:
No, and the missing `#include <stdio.h>` appears to be a bug in libjpeg, surprisingly enough. I'll have to investigate it a bit further.

Reply:
@RubenRBS Indeed, `libjpeg` itself is buggy. #23428

Reply:
Nope. it's apparently by design for libjpeg. I'll try to submit a patch upstream for Netpbm.

Reply:
Thanks for following up on it, really appreciated! Let me know once this has been reported upstream :)